Atomic/sign.py

```diff
@@ -50,14 +46,46 @@ def sign(self):
-                                                     os.path.basename(expanded_image_name), manifest_hash)
-                self.make_sig_dirs(sigstore_path)
-                sig_name = self.get_sig_name(sigstore_path)
-                fq_sig_path = os.path.join(sigstore_path, sig_name)
-                if os.path.exists(fq_sig_path):
-                    raise ValueError("The signature {} already exists.  If you wish to "
-                                     "overwrite it, please delete this file first")
+
+                if self.args.signature_path:
+                    if not os.path.exists(self.args.signature_path):
+                        raise ValueError("The path {} does not exist".format(self.args.signature_path))
+                    fq_sig_path = os.path.join(self.args.signature_path,
+                                               self.get_sig_name(self.args.signature_path))
+
+                else:
+                    reg, repo, _ = util.decompose(expanded_image_name)
+                    reg_info = util.have_match_registry("{}/{}".format(reg, repo), registry_configs)
+                    # No default and no registry match, exit
+                    if reg_info is None and default_store is None:
+                        raise ValueError("Unable to associate {} with "
+                                         "configurations in {} and no 'default-docker' "
+                                         "is defined.".format(sign_image, registry_config_path))
+                    # If no registry match but there is a default defined, use it
+                    elif reg_info is None and default_store is not None:
+                        reg_info = default_store
+
+                    signature_path = util.get_signature_write_path(reg_info)
+                    if signature_path is None:
+                        raise ValueError("No write path for {}/{} was "
+                                         "found in {}".format(reg, repo, registry_config_path))
+
+                    # Deal with write path prepends
+                    if signature_path.startswith("file://"):
+                        signature_path = signature_path.replace("file://", "")
+
+                        # Make sure signature path exists
+                        if not os.path.exists(signature_path):
+                            raise ValueError("The signature path {} does not exist".format(signature_path))
+
+                    sigstore_path = "{}/{}/{}@{}".format(signature_path, os.path.dirname(expanded_image_name),
+                                                         os.path.basename(expanded_image_name), manifest_hash)
+                    self.make_sig_dirs(sigstore_path)
+                    sig_name = self.get_sig_name(sigstore_path)
+                    fq_sig_path = os.path.join(sigstore_path, sig_name)
+                    if os.path.exists(fq_sig_path):
```

Comment:
Nope, this is the earliest.  This hasn't changed since the first submission.  We need to collection all the information including path and signature name until we can run this check.